### PR TITLE
feat: added auth middleware

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**

### CODE REVIEW

**Review by asafMasa — APPROVED** [review]

File: authFilter.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    def __call__(self, environ, start_response):
+        # put authorize callback function into environment
+        environ['mapproxy.authorize'] = self.authorize
+        return self.app(environ, start_response)
+
+    def authorize(self, service, layers=[], environ=None, **kw):
+        if service.startswith('wms.'):
+            token = environ.get(f'HTTP_{self.autHeaderName}') if(self.autHeaderName) else None
+            if(token == None and self.upperAuthQueryName):
```
Comment (ronenkapelian) [inline authFilter.py:26]: more likely pythonic way is write "!token" that none is falsy value

File: authFilter.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                query = parse_qs(environ['QUERY_STRING'])
+                token = query.get(self.upperAuthQueryName,query.get(self.lowerAuthQueryName,[None]))[0]
```
Comment (ronenkapelian) [inline authFilter.py:28]: more readble and likley to define line before the default value as parameters to make it more clearly

File: authFilter.py (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply authFilter.py:26]: the pylance (linter) claims that there is no such thing.
! is also not in python operators list

File: authFilter.py (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply authFilter.py:28]: what do you mean?

**Review by ronenkapelian — APPROVED** [review]

**Pull request merged**